packages/cli/src/commands/pack.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { paths } from '../lib/paths';
+
+export const pre = async () => {
+  const pkgPath = paths.resolveTarget('package.json');
+
+  const pkg = await fs.readJson(pkgPath);
+  pkg.types = 'dist/index.d.ts';
+  await fs.writeJson(pkgPath, pkg, { encoding: 'utf8', spaces: 2 });
+};
+
+export const post = async () => {
+  const pkgPath = paths.resolveTarget('package.json');
+
+  const pkg = await fs.readJson(pkgPath);
+  pkg.types = 'src/index.ts';
```

Comment:
I guess this could have been served a bit cleaner as doing a backup in a secret dotfile, and restoring from that backup. That would not give unintended changes if the user has a slightly different thing going on in their file (other indent or whatever)

Reply:
And I presume there was no way to make this happen silently inside build? It would just be nice to not have so much boilerplate in the package files.

Reply:
yeah opted for just making sure type field exists everywhere instead of backup/restore. That would've been a bit annoying to deal with too imo, especially since the postpack script doesn't work :grin:.

And no, afaik there's no way to silently hook into the process. I mean we could modify the yarn implementation we check into the repo, but nooooooooooo x)